packages/components/src/lib/tableProvider.ts

```diff
@@ -9,7 +11,9 @@ export function parquetDataFrame(from: AsyncBufferFrom, metadata: FileMetaData):
   const { children } = parquetSchema(metadata)
   const header = children.map(child => child.element.name)
   const sortCache = new Map<string, Promise<number[]>>()
-  const data = new Array(Number(metadata.num_rows))
+  const data = new Array<ResolvableRow>(Number(metadata.num_rows))
+  // TODO(SL) ^ isn't this too big?
+  // TODO(SL) ^ the type is currently a lie, because all rows are undefined for now
```

Comment:
In practice I haven't found this to be a problem. Worst case is (roughly) `num_rows * pointer_size` but most ML datasets are 100s of thousands to low million rows per parquet file. So only a few mb of array. Also I'm not sure but the js engine might lazy init the array. It feels fast and haven't seen it out-of-memory error here.

Fair point on the type is a lie.

Reply:
OK. I changed the comments.